src/clients/java/src/main/java/com/tigerbeetle/NativeClient.java

```diff
@@ -36,55 +108,29 @@ private static void assertArgs(final byte[] clusterID, final String addresses,
     }
 
     private NativeClient(final long contextHandle) {
-        this.contextHandle = new AtomicLong(contextHandle);
-        this.contextHandleReferences = new AtomicLong(0L);
+        try {
+            this.handle = new NativeHandle(contextHandle);
+            this.cleanable = cleaner.register(this, handle);
+        } catch (Throwable forward) {
+            clientDeinit(contextHandle);
+            throw forward;
+        }
```

Comment:
Isn't this try-finally ?

Reply:
Ahh, it's errdefer, not defer :facepalm:

Reply:
I can't live without `errdefer` anymore ❤️